core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,131 @@
+
+/** @packageDocumentation
+ * @module Curve
+ */
+
+/** Data carrier used by the sampler.
+ * @internal
+*/
+export class QuadrantFractions {
+  private _quadrant: 0|1|2|3;
+  private _fractions: number[];
+  private constructor(quadrant: 0|1|2|3, fractions?: number[]) {
+    this._quadrant = quadrant;
+    this._fractions = fractions ? fractions : [];
+  }
+  public create(quadrant: number, fractions?: number[]): QuadrantFractions {
+    return new QuadrantFractions(Math.abs(quadrant) % 4 as 0|1|2|3, fractions);
+  }
+  public get quadrant(): number {
+    return this._quadrant + 1;
+  }
+  public get fractions(): number[] {
+    return this._fractions;
+  }
+};
+
+/**
+ * Context for sampling an elliptical Arc3d, e.g., for approximations.
+ * @internal
+ */
+export class EllipticalArcApproximationContext {
+  private _arc: Arc3d;
+  private _axx: number;
+  private _ayy: number;
+  private _curvatureRange: Range1d;
+  private _isValidArc: boolean;
+
+  public constructor(arc: Arc3d) {
+    const scaleData = arc.toScaledMatrix3d();
+    this._arc = Arc3d.createScaledXYColumns(scaleData.center, scaleData.axes, scaleData.r0, scaleData.r90, scaleData.sweep);
+    this._axx = arc.matrixRef.columnXMagnitudeSquared();
+    this._ayy = arc.matrixRef.columnYMagnitudeSquared();
+    this._curvatureRange = Range1d.createNull();
+    this._isValidArc = true;
+
+    if (Geometry.isSmallMetricDistanceSquared(this._axx) || Geometry.isSmallMetricDistanceSquared(this._ayy))
+      this._isValidArc = false; // ellipse must have positive radii
+    else if (Geometry.isSameCoordinateSquared(this._axx, this._ayy))
+      this._isValidArc = false; // ellipse must not be circular
+    else {
+      // extreme curvatures are at the semi-axis points
+      this._curvatureRange.extendX(Math.sqrt(this._axx) / this._ayy);
+      this._curvatureRange.extendX(Math.sqrt(this._ayy) / this._axx);
+    }
+  }
+
+  public get arc(): Arc3d {
+    return this._arc;
+  }
+  public get vector0LengthSquared(): number {
+    return this._axx;
+  }
+  public get vector90LengthSquared(): number {
+    return this._ayy;
+  }
+  public get curvatureRange(): Range1d {
+    return this._curvatureRange;
+  }
+  public get isValidArc(): boolean {
+    return this._isValidArc;
+  }
+
+  /**
+   * Compute the angle corresponding to the point in the ellipse's first quadrant with the given curvature.
+   * * The instance sweep is ignored.
+   * * Arc is assumed to be non-circular and have perpendicular axes of positive length.
+   * * This is a scaled inverse of [[Arc3d.fractionToCurvature]] restricted to fractions in [0,1/4].
+   * @return radian angle in [0,pi/2] or undefined if the ellipse is invalid, or does not attain the given curvature.
+  */
+  private curvatureToAngle(curvature: number): number | undefined {
+    if (!this.isValidArc)
+      return undefined;
+    if (!this.curvatureRange.containsX(curvature))
+      return undefined; // ellipse does not attain this curvature
```

Comment:
I think we should document the math details of lines 93-97.

From Eberly PDF we have `x = a*sqrt( |lambda - a^2| / |b2 - a^2| )` where `lambda = (a*b / K )^(3/2)` so `cos(theta) = x/a = sqrt( |lambda - a^2| / |b2 - a^2| )` which is calculated here. We can add this as a comment to the function body but still derivation of the original formula we used for `x` is not clear. Therefore, I think we should add math details to an internal md file. I can do that.